wrangler/src/@utils/flush-single-use-auth-tokens.js

```diff
@@ -0,0 +1,44 @@
+import { response } from 'cfw-easy-utils'
+import moment from 'moment'
+import { groupBy, map } from 'lodash'
+import { handleResponse } from './fetch'
+import Bluebird from 'bluebird'
+
+export default async function flushSingleUseAuthTokens({ env }) {
+  const { META, TX_FEES } = env
+
+  const { keys } = await META.list({prefix: 'suat:', limit: 100})
+
+  const allExpiredKeys = keys
+  .filter(({metadata}) => moment.utc(metadata, 'X').isBefore())
+  .map(({name}) => {
+    const [, publicKey, transactionHash] = name.split(':')
```

Comment:
does CF have an expiring KV store? I'm not sure were as subject to the parallel requirements here.

Sure, if we use basic KV, we might let 2-3 slip if they are submitted at the same time, but could be worth considering if the cron job ends up being problematic.

I'm happy with the way it is, just some food for thought.

Reply:
They do have an expiring KV but not an expiring DO store (yet). Rumor has it something like this is in the works. For now it's a combination to allow instant db puts/gets with KV to ensure we get the DO store flushed of old tokens.

We can't use KV alone as I could spam concurrent requests with the "singleUse" token and as long as A) the key didn't exist in the KV store (potential 1 minute latency) and the token was still valid via the `maxTime` the request would go through. There's not really a limit on current requests so the 2-3 becomes hundreds, thousands, millions?? A 1 hour token + 1 minute latency is a lot of requests.

Reply:
Fair point! Looks like the DO approach is fairly light too, so I am happy with it.